PlagiarismSettingsForm.inc.php

```diff
@@ -59,9 +117,41 @@ function fetch($request, $template = null, $display = false) {
 	/**
 	 * @copydoc Form::execute()
 	 */
-	function execute(...$functionArgs) {
-                $this->_plugin->updateSetting($this->_contextId, 'ithenticateUser', trim($this->getData('ithenticateUser'), "\"\';"), 'string');
-		$this->_plugin->updateSetting($this->_contextId, 'ithenticatePass', trim($this->getData('ithenticatePass'), "\"\';"), 'string');
+	public function execute(...$functionArgs) {
+		
+		$ithenticateApiUrl = trim($this->getData('ithenticateApiUrl'), "\"\';");
+		$ithenticateApiKey = trim($this->getData('ithenticateApiKey'), "\"\';");
+
+		if (!empty(array_filter([$ithenticateApiUrl, $ithenticateApiKey]))) {
+
+			if ($this->_plugin->getSetting($this->_context->getId(), 'ithenticateApiUrl') !== $ithenticateApiUrl ||
+				$this->_plugin->getSetting($this->_context->getId(), 'ithenticateApiKey') !== $ithenticateApiKey) {
+				
+				// access updated or new access entered, need to update webhook registration	
+				$this->_plugin->registerIthenticateWebhook(
```

Comment:
How is the webhook updated when there's a configuration change in `config.inc.php`?

Reply:
Great catch . Yes, right now on the update of `config.inc.php` credentials, we have no way to update the webhook . we can do one of the following 
1. before each file upload , check the validity of currently store webhook details, not very ideal and too much API request
2. have a schedular task to validate/verify/update webhook in the plugin, a lot of work
3. Have a button in plugin settings form to update webhook details manually, easier but confusing behaviour . 

Any other suggestion aside from those ?

Reply:
Changing the configuration is going to be an extremely rare operation so I don't think it's worth many lines of code devoted to it. We could simply add a note to the configuration section that changing the credential will invalidate any existing web hooks. Will resubmitting the file to iThenticate through the web interface cause a new web hook to be generated?

Reply:
> Will resubmitting the file to iThenticate through the web interface cause a new web hook to be generated?

It can but as same as No.1 , on each file submission we need verify the current webhook details . And configuring and registering the webhook also implemented when uploading the files from web interface . 

Basically what we do is at the very first submission, check if the webhook configured for context and if not, register it and store details in context settings table . If those details available, we assume webhook configured and registered . Updating the config file creds will not remove those details so same result .

Reply:
I would suggest adding a warning to the configuration template in `config.inc.php` and not worrying about it otherwise. Changing the credentials mid-stream is a pretty unusual operation.